Content.Shared/Sound/SharedEmitSoundSystem.cs

```diff
@@ -99,6 +102,14 @@ private void OnEmitSoundOnUseInHand(EntityUid uid, EmitSoundOnUseComponent compo
             args.Handled = true;
     }
 
+    private void OnEmitSoundOnAction(EntityUid uid, EmitSoundOnActionComponent component, SoundActionEvent args)
+    {
+        TryEmitSound(uid, component, args.Performer);
+
+        if (component.Handle)
+            args.Handled = true;
+    }
```

Comment:
This format of subscribing to an event is preferred, it associates the comp with the entity and is always handled by ref
```suggestion
    private void OnEmitSoundOnAction(Entity<EmitSoundOnActionComponent> ent, ref SoundActionEvent args)
    {
        TryEmitSound(ent, ent, args.Performer);

        if (ent.Comp.Handle)
            args.Handled = true;
    }
```